clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy

```diff
@@ -96,51 +97,81 @@ class UpsertAzureAppGatewayAtomicOperation implements AtomicOperation<Map> {
         errList = AzureDeploymentOperation.checkDeploymentOperationStatus(task, BASE_PHASE, description.credentials, resourceGroupName, deployment.name)
       } else {
         // We are attempting to create a new application gateway
-        description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, virtualNetworkName, description.region)
+        if (!description.useDefaultVnet) {
+          task.updateStatus(BASE_PHASE, "Create ApplicationGateway using virtual network $description.vnet and subnet $description.subnet for server group $description.name")
 
-        task.updateStatus(BASE_PHASE, "Creating subnet for application gateway")
+          // Create corresponding ResourceGroup if it's not created already
+          description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, null, description.region)
```

Comment:
By the way, there is call to this method, _initializeResourceGroupAndVNet_, that occurs when we invoke the template deployment method, _createResourceFromTemplate_. Might be worth looking into removing that call so that we don't inadvertently create another vNet when we're using the one selected by the user.